### PR TITLE
feat: adds opened and closed status icon to Block and updates UI of related components

### PR DESCRIPTION
**Pull request by asangma**
**Related Issue:** #1542 

## Summary
Adds an icon that indicates opened or closed status of the Block.
Only adds the icon if the Block is collapsible.

Also updates the UI of other components to both increase alignment and distinguish functionality.

@mitc7862 @bstifle Could use your UI review as well.

View demos/shell/block-configurations.html to see permutations of the Block header and how it relates to Panel's back button and BlockSection.

cc @AdelheidF

### CODE REVIEW

File: src/components/calcite-block/calcite-block.tsx [hunk 1]
```diff
@@ -164,7 +171,9 @@ export class CalciteBlock {
     const hasControl = getSlotted(el, SLOTS.control);
 
     const headerNode = (
-      <div class={CSS.headerContainer}>
+      <div
```
Comment (driskull) [inline src/components/calcite-block/calcite-block.tsx:174]: nitpick: run prettier :)

File: src/components/calcite-block/calcite-block.tsx [hunk 2]
```diff
@@ -145,13 +145,20 @@ export class CalciteBlock {
...
+
     const headerContent = (
       <header class={CSS.header}>
         {hasIcon ? (
           <div class={CSS.icon}>
             <slot name={SLOTS.icon} />
           </div>
         ) : null}
+        {collapsible ? (
+          <div class={CSS.toggleIcon} aria-hidden="true">
```
Comment (driskull) [inline src/components/calcite-block/calcite-block.tsx:158]: nitpick: Is this div needed or could the class and aria-hidden be applied directly to the calcite-icon?

**Review by driskull — APPROVED** [review]
👍

File: src/components/calcite-block/calcite-block.tsx (hunk 1 above)
Comment (asangma) [reply src/components/calcite-block/calcite-block.tsx:174]: 🧹

File: src/components/calcite-block/calcite-block.tsx (hunk 2 above)
Comment (asangma) [reply src/components/calcite-block/calcite-block.tsx:158]: yeah...prolly.

**Comment by asangma** [comment]
Had some updates after some design reviews. Please give it another once over. 🙏🏾 @driskull

File: src/components/calcite-block/calcite-block.tsx [hunk 3]
```diff
@@ -1,5 +1,5 @@
 import { Component, Element, Event, EventEmitter, Host, Prop, h, VNode } from "@stencil/core";
-import { CSS, SLOTS, TEXT, HEADING_LEVEL } from "./resources";
+import { CSS, SLOTS, TEXT, HEADING_LEVEL, TOGGLE_ICON } from "./resources";
```
Comment (driskull) [inline src/components/calcite-block/calcite-block.tsx:2]: nitpick: We usually call this const `ICONS`.

**Review by driskull — APPROVED** [review]
👍

**Comment by julio8a** [comment]
Did @bstifle and @macandcheese look at this?

**Comment by macandcheese** [comment]
> Did @bstifle and @macandcheese look at this?

Yep, layout looks great as of last updates

**Comment by bstifle** [comment]
Yep, worked with alan a bit on this. looks goodly

File: src/components/calcite-block/calcite-block.tsx (hunk 3 above)
Comment (asangma) [reply src/components/calcite-block/calcite-block.tsx]: Thanks @driskull. Updated.

**Comment by asangma** [comment]
Thanks y'all. Murgling!

**Pull request merged**